src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -136,6 +125,17 @@ struct program_node {
     virtual primitive_type_id type() const { return desc->type; }
     virtual std::shared_ptr<kernel_selector::fuse_params> get_fuse_params() const { return nullptr; }
 
+    virtual std::shared_ptr<kernel_impl_params> get_kernel_impl_params() const {
+        return get_kernel_impl_params(get_input_layouts(), output_layout);
+    }
+
+    virtual std::shared_ptr<kernel_impl_params> get_kernel_impl_params(const std::vector<layout>& in_layouts,
```

Comment:
Do we need to return `shared_ptr` here? Can it be replaced with `unique_ptr`?

Reply:
I thinks its possible. But will we get more benefit?

Reply:
It's just a matter of clear design. If we don't really have shared ownership for these objects, then `unique_ptr` would be better.

Reply:
Fixed.